test/kernels/test_deprecated_qkernel_methods.py

```diff
@@ -0,0 +1,709 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# NOTE WELL that this file of tests should disappear when we remove the
+# deprecated ``user_param`` argument/property/methods in quantum_kernel.py.
+# These tests exist ONLY to ensure that these deprecated objects correctly.
+
+""" Test QuantumKernel """
```

Comment:
Can we subset this down to just testing the user_parameter renaming related logic. Ie that using the deprecated parameter and the deprecated properties/methods work as expected. This, I would think, would be more just the original tests that were changed over to the new naming. 
 
Also when we test function that we expect to raise deprecated warnings we also suppress these. If you look in the CI results we otherwise collect all deprecated messages so as to ensure we are not ourselves using deprecated logic in the code which we need to fix. So we avoid having messages from CI when we know we are explicitly calling such deprecated logic just to test it.

Reply:
1. oh, aha, yes, I'll suppress the deprecation warnings

2. I don't understand what you're suggesting:
`This, I would think, would be more just the original tests that were changed over to the new naming.`
Do you mean that I should just copy over the original test-file and rename the few places to get it working, so it will test that the originally-named(-now-deprecated) methods still work?  B/c that's what I did.  Or do you mean that you'd like to see just a few tests for these deprecated methods?  The reason I chose to do it this way, is that it is guaranteed to test all the relevant behaviour, and once the deprecated methods are removed, we can just remove the entire test-file.

The `test_deprecated.....py` file is literally "start with test_qkernel.py and rename the few spots that need renaming so it'll run, and leave all the deprecated method-names in-place unmodified".

Does that convince?

Reply:
I think for part 2 I was thinking all that was needed was just those specific tests that you needed to update with the new names. Here they would be just as they were with the old names. I think this represents quite a significant sub-set of all the tests in this file. If the test does not explicitly deal with the old deprecated names/properties I think we should be able to rely on the original updated file to cover everything right?

Reply:
It should just be a copy of those tests from the original file that were renamed right - at least in theory. I did not check in detail but was hoping they would be easy enough to isolate.  Yes the file will go away but ahead of then it will increase the CI time by doubling up on all the other tests - it will be around at least 3 months and more likely 6 or more given we tend to release roughly quarterly or so.

Reply:
Fair enough: I'm removing everything that doesn't concern the renaming.